httpbp/middlewares.go

```diff
@@ -108,21 +110,32 @@ func StartSpanFromTrustedRequest(
 	return tracing.StartSpanFromHeaders(ctx, name, spanHeaders)
 }
 
+func HTTPErrorSuppressor(err error) bool {
+	var httpErr HTTPError
+	if errors.As(err, &httpErr) {
+		return httpErr.Response().Code < 500
+	}
+	return false
+}
+
 // InjectServerSpan returns a Middleware that will automatically wrap the
 // HansderFunc in a new server span and stop the span after the function
 // returns.
 //
 // InjectServerSpan should generally not be used directly, instead use the
 // NewBaseplateServer function which will automatically include InjectServerSpan
 // as one of the Middlewares to wrap your handlers in.
-func InjectServerSpan(truster HeaderTrustHandler) Middleware {
+func InjectServerSpan(truster HeaderTrustHandler, suppressor errorsbp.Suppressor) Middleware {
```

Comment:
this is a breaking change. do we have a way to do it non-breaking? I think we can just always use `HTTPErrorSuppressor` in this version (don't change the API, but change the behavior slightly), and allow customization in a follow-up PR that's a breaking change (and we'll wait to merge that breaking change PR).